pep/app/pods/components/document/text/component.ts

```diff
@@ -261,6 +257,25 @@ export default class DocumentText extends Component<DocumentTextArgs> {
             if (targetSearchHit) {
                 this.scrollToSearchHit(targetSearchHit);
             }
+        } else if (type === DocumentLinkTypes.BANNER) {
+            const parent = target.parentElement;
+            if (parent) {
+                const sourceCode = parent.getAttribute('data-journal-code');
+                if (sourceCode) {
+                    this.router.transitionTo('browse.journal', sourceCode);
+                }
+            }
+        } else if (type === DocumentLinkTypes.TITLE) {
+            const parent = target.parentElement;
+            if (parent) {
+                const sourceCode = parent.getAttribute('data-journal-code');
+                const volume = parent.getAttribute('data-volume');
+                const page = parent.getAttribute('data-page');
+
+                if (sourceCode && volume) {
+                    this.router.transitionTo('browse.journal.volume', sourceCode, volume);
+                }
+            }
```

Comment:
I think i mentioned this in an older PR but this onDocumentClick action and if block is just getting ginormous. Would be nice if we could chunk this out into like a hash that maps to a bunch of separate helper methods, e.g. something like

```ts
const documentClickActions = {
   [DocumentLinkTypes.GLOSSARY_TERM]: 'viewGlossaryTerm',
   [DocumentLinkTypes.AUTHOR_SEARCH]: 'viewAuthor',
   ...
}
```
and then this method just gets reduced to mainly just calling the correct method from that hash essentially:

```ts
onDocumentClick(event: Event) {
    //common stuff here...
   this[documentClickActions[type]](target);
}
```